core/lib/dal/src/proof_generation_dal.rs

```diff
@@ -129,7 +129,7 @@ impl ProofGenerationDal<'_, '_> {
         let result: Option<L1BatchNumber> = sqlx::query!(
             "SELECT l1_batch_number \
              FROM proof_generation_details \
-             WHERE status <> 'generated' \
+             WHERE status = 'picked_by_prover' \
```

Comment:
what if we don't have any rows that are "picked by prover" for some reason? Same applies to the query above.

Reply:
anyway, currently this and the above methods have confusing names and it's not clear for me what exactly they report. Why does oldest-not-generated-batch has to be `picked_by_prover`? What if the prover is down, and we don't move statuses to `picked_by_prover` at all?

I was thinking the previous query was correct - can you please explain why it's not? Only because `skipped`? If so, we can do `NOT IN`

Reply:
Yes, agree about names, and yes, it is not correct because of `skipped`. Talking about `picked_by_prover` status, I guess I didn't really thought about it, it will return incorrect values if the scenario you described will occur.

Reply:
But I don't think that the same thing can be applied to query above